Compilation issues on Grace Hopper

## Expected Behavior

HPX schould compile with gcc 12 or gcc 13.

## Actual Behavior
 
I get the following error:

```
212 errors found in build log:
     4274    
     4275    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/parallel/util/detail/chunk
             _size_iterator.hpp(350): warning #20012-D: __device__ annotation is ignored on a function("chunk_size_idx_iterator") that is explicitly defaulted on it
             s first declaration
     4276             __attribute__((host)) __attribute__((device))
     4277                                                  ^
     4278    
     4279    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp: In fun
             ction 'void hpx::util::detail::format_value(std::ostream&, std::string_view, const format_join<Range>&)':
  >> 4280    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:34:
              error: '__T0' was not declared in this scope; did you mean '__y0'?
     4281      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4282          |                                  ^~~~
     4283          |                                  __y0
  >> 4284    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:38:
              error: template argument 1 is invalid
     4285      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4286          |                                      ^
  >> 4287    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 1 is invalid
     4288      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4289          |                                                 ^
  >> 4290    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 2 is invalid
  >> 4291    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:51:
              error: '<declaration error>' is not a template [-fpermissive]
     4292      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4293          |                                                   ^~
  >> 4294    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:73:
              error: expression list treated as compound expression in initializer [-fpermissive]
     4295      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4296          |                                                                         ^
     4297    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp: In fun
             ction 'void hpx::util::detail::format_value(std::ostream&, std::string_view, const format_join<Range>&)':
  >> 4298    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:34:
              error: '__T0' was not declared in this scope; did you mean '__y0'?
     4299      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4300          |                                  ^~~~
     4301          |                                  __y0
  >> 4302    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:38:
              error: template argument 1 is invalid
     4303      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4304          |                                      ^
  >> 4305    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 1 is invalid
     4306      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4307          |                                                 ^
  >> 4308    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 2 is invalid
  >> 4309    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:51:
              error: '<declaration error>' is not a template [-fpermissive]
     4310      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4311          |                                                   ^~
  >> 4312    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:73:
              error: expression list treated as compound expression in initializer [-fpermissive]
     4313      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4314          |                                                                         ^
     4315    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp: In fun
             ction 'void hpx::util::detail::format_value(std::ostream&, std::string_view, const format_join<Range>&)':
  >> 4316    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:34:
              error: '__T0' was not declared in this scope; did you mean '__y0'?
     4317      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4318          |                                  ^~~~
     4319          |                                  __y0
  >> 4320    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:38:
              error: template argument 1 is invalid
     4321      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4322          |                                      ^
  >> 4323    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 1 is invalid
     4324      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4325          |                                                 ^
  >> 4326    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 2 is invalid
  >> 4327    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:51:
              error: '<declaration error>' is not a template [-fpermissive]
     4328      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4329          |                                                   ^~
  >> 4330    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:73:
              error: expression list treated as compound expression in initializer [-fpermissive]
     4331      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4332          |                                                                         ^
     4333    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'decltype (rebind_container<hpx::util::detail::container_remapping::mapped_type_from_t<T, M> >(container)) hpx::util::
             detail::container_remapping::remap_container(container_mapping_tag<false, false>, M&&, T&&)':
  >> 4334    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:518:94: error: '__T0' does not name a type
     4335      518 |                 remapped.push_back(spreading::unpack(
     4336          |                                                                                              ^
     4337    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'std::decay_t<U> hpx::util::detail::container_remapping::remap_container(container_mapping_tag<false, true>, M&&, T&&)
             ':
  >> 4338    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:533:81: error: '__T0' does not name a type
     4339      533 |                 val = spreading::unpack(
     4340          |                                                                                 ^
     4341    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'hpx::util::detail::spreading::spread_box<> hpx::util::detail::container_remapping::remap_container(container_mapping_
             tag<true, false>, M&&, T&&)':
  >> 4342    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:548:57: error: '__T0' does not name a type
     4343      548 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(val), val));
     4344          |                                                         ^~~~
     4345    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'void hpx::util::detail::container_remapping::remap(hpx::util::detail::strategy_traverse_tag, T&&, M&&, std::enable_if
             _t<is_effective_v<M, typename std::conditional<is_rvalue_reference_v<Container&&>, decltype (static_cast<typename std::remove_reference<decltype (* dec
             lval<Container>().begin())>::type&&>(* declval<Container>().begin())), decltype (* declval<Container>().begin())>::type> >*)':
  >> 4346    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:574:57: error: '__T0' does not name a type
     4347      574 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(element), element));
     4348          |                                                         ^~~~
     4349    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'decltype (rebind_container<hpx::util::detail::container_remapping::mapped_type_from_t<T, M> >(container)) hpx::util::
             detail::container_remapping::remap_container(container_mapping_tag<false, false>, M&&, T&&)':
  >> 4350    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:518:94: error: '__T0' does not name a type
     4351      518 |                 remapped.push_back(spreading::unpack(
     4352          |                                                                                              ^
     4353    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'std::decay_t<U> hpx::util::detail::container_remapping::remap_container(container_mapping_tag<false, true>, M&&, T&&)
             ':
  >> 4354    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:533:81: error: '__T0' does not name a type
     4355      533 |                 val = spreading::unpack(
     4356          |                                                                                 ^
     4357    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'hpx::util::detail::spreading::spread_box<> hpx::util::detail::container_remapping::remap_container(container_mapping_
             tag<true, false>, M&&, T&&)':
  >> 4358    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:548:57: error: '__T0' does not name a type
     4359      548 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(val), val));
     4360          |                                                         ^~~~
     4361    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'void hpx::util::detail::container_remapping::remap(hpx::util::detail::strategy_traverse_tag, T&&, M&&, std::enable_if
             _t<is_effective_v<M, typename std::conditional<is_rvalue_reference_v<Container&&>, decltype (static_cast<typename std::remove_reference<decltype (* dec
             lval<Container>().begin())>::type&&>(* declval<Container>().begin())), decltype (* declval<Container>().begin())>::type> >*)':
  >> 4362    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:574:57: error: '__T0' does not name a type
     4363      574 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(element), element));
     4364          |                                                         ^~~~
     4365    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'decltype (rebind_container<hpx::util::detail::container_remapping::mapped_type_from_t<T, M> >(container)) hpx::util::
             detail::container_remapping::remap_container(container_mapping_tag<false, false>, M&&, T&&)':
  >> 4366    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:518:94: error: '__T0' does not name a type
     4367      518 |                 remapped.push_back(spreading::unpack(
     4368          |                                                                                              ^
     4369    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'std::decay_t<U> hpx::util::detail::container_remapping::remap_container(container_mapping_tag<false, true>, M&&, T&&)
             ':
  >> 4370    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:533:81: error: '__T0' does not name a type
     4371      533 |                 val = spreading::unpack(
     4372          |                                                                                 ^
     4373    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'hpx::util::detail::spreading::spread_box<> hpx::util::detail::container_remapping::remap_container(container_mapping_
             tag<true, false>, M&&, T&&)':
  >> 4374    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:548:57: error: '__T0' does not name a type
     4375      548 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(val), val));
     4376          |                                                         ^~~~
     4377    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'void hpx::util::detail::container_remapping::remap(hpx::util::detail::strategy_traverse_tag, T&&, M&&, std::enable_if
             _t<is_effective_v<M, typename std::conditional<is_rvalue_reference_v<Container&&>, decltype (static_cast<typename std::remove_reference<decltype (* dec
             lval<Container>().begin())>::type&&>(* declval<Container>().begin())), decltype (* declval<Container>().begin())>::type> >*)':
  >> 4378    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:574:57: error: '__T0' does not name a type
     4379      574 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(element), element));
     4380          |                                                         ^~~~
     4381    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp: In lambda function:
  >> 4382    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:84: error: '__T0' was not declared in this scope; did you mean '__y0'?
     4383      273 |                             return hpx::execution::experimental::connect(
     4384          |                                                                                    ^
     4385          |                                                                                    __y0
  >> 4386    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:88: error: template argument 1 is invalid
     4387      273 |                             return hpx::execution::experimental::connect(
     4388          |                                                                                        ^
  >> 4389    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:91: error: expected '>' before '&&' token
     4390      273 |                             return hpx::execution::experimental::connect(
     4391          |                                                                                           ^
  >> 4392    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:91: error: expected '(' before '&&' token
  >> 4393    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:93: error: expected identifier before '>' token
     4394      273 |                             return hpx::execution::experimental::connect(
     4395          |                                                                                             ^
     4396    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp: In member function 'void hpx::when_all_vector_detail::when_all_vector_sender_impl<Sender>::when_all_vector_sender_type::operation_state
             <Receiver>::finish()':
  >> 4397    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:56: error: '__T0' was not declared in this scope; did you mean '__y0'?
     4398      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4399          |                                                        ^~~~
     4400          |                                                        __y0
  >> 4401    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:60: error: template argument 1 is invalid
     4402      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4403          |                                                            ^
  >> 4404    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:63: error: expected '>' before '&&' token
     4405      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4406          |                                                               ^~
  >> 4407    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:63: error: expected '(' before '&&' token
  >> 4408    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:65: error: expected identifier before '>' token
     4409      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4410          |                                                                 ^
     4411    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp: In fun
             ction 'void hpx::util::detail::format_value(std::ostream&, std::string_view, const format_join<Range>&)':
  >> 4412    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:34:
              error: '__T0' was not declared in this scope; did you mean '__y0'?
     4413      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4414          |                                  ^~~~
     4415          |                                  __y0
  >> 4416    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:38:
              error: template argument 1 is invalid
     4417      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4418          |                                      ^
  >> 4419    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 1 is invalid
     4420      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4421          |                                                 ^
  >> 4422    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 2 is invalid
  >> 4423    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:51:
              error: '<declaration error>' is not a template [-fpermissive]
     4424      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4425          |                                                   ^~
  >> 4426    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:73:
              error: expression list treated as compound expression in initializer [-fpermissive]
     4427      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4428          |                                                                         ^
     4429    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp: In lambda function:
  >> 4430    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:84: error: '__T0' was not declared in this scope; did you mean '__y0'?
     4431      273 |                             return hpx::execution::experimental::connect(
     4432          |                                                                                    ^
     4433          |                                                                                    __y0
  >> 4434    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:88: error: template argument 1 is invalid
     4435      273 |                             return hpx::execution::experimental::connect(
     4436          |                                                                                        ^
  >> 4437    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:91: error: expected '>' before '&&' token
     4438      273 |                             return hpx::execution::experimental::connect(
     4439          |                                                                                           ^
  >> 4440    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:91: error: expected '(' before '&&' token
  >> 4441    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:93: error: expected identifier before '>' token
     4442      273 |                             return hpx::execution::experimental::connect(
     4443          |                                                                                             ^
     4444    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp: In member function 'void hpx::when_all_vector_detail::when_all_vector_sender_impl<Sender>::when_all_vector_sender_type::operation_state
             <Receiver>::finish()':
  >> 4445    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:56: error: '__T0' was not declared in this scope; did you mean '__y0'?
     4446      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4447          |                                                        ^~~~
     4448          |                                                        __y0
  >> 4449    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:60: error: template argument 1 is invalid
     4450      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4451          |                                                            ^
  >> 4452    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:63: error: expected '>' before '&&' token
     4453      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4454          |                                                               ^~
  >> 4455    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:63: error: expected '(' before '&&' token
  >> 4456    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:65: error: expected identifier before '>' token
     4457      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4458          |                                                                 ^
     4459    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp: In lambda function:
  >> 4460    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:84: error: '__T0' was not declared in this scope; did you mean '__y0'?
     4461      273 |                             return hpx::execution::experimental::connect(
     4462          |                                                                                    ^
     4463          |                                                                                    __y0
  >> 4464    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:88: error: template argument 1 is invalid
     4465      273 |                             return hpx::execution::experimental::connect(
     4466          |                                                                                        ^
  >> 4467    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:91: error: expected '>' before '&&' token
     4468      273 |                             return hpx::execution::experimental::connect(
     4469          |                                                                                           ^
  >> 4470    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:91: error: expected '(' before '&&' token
  >> 4471    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:93: error: expected identifier before '>' token
     4472      273 |                             return hpx::execution::experimental::connect(
     4473          |                                                                                             ^
     4474    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp: In member function 'void hpx::when_all_vector_detail::when_all_vector_sender_impl<Sender>::when_all_vector_sender_type::operation_state
             <Receiver>::finish()':
  >> 4475    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:56: error: '__T0' was not declared in this scope; did you mean '__y0'?
     4476      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4477          |                                                        ^~~~
     4478          |                                                        __y0
  >> 4479    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:60: error: template argument 1 is invalid
     4480      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4481          |                                                            ^
  >> 4482    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:63: error: expected '>' before '&&' token
     4483      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4484          |                                                               ^~
  >> 4485    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:63: error: expected '(' before '&&' token
  >> 4486    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:65: error: expected identifier before '>' token
     4487      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4488          |                                                                 ^
     4489    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'decltype (rebind_container<hpx::util::detail::container_remapping::mapped_type_from_t<T, M> >(container)) hpx::util::
             detail::container_remapping::remap_container(container_mapping_tag<false, false>, M&&, T&&)':
  >> 4490    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:518:94: error: '__T0' does not name a type
     4491      518 |                 remapped.push_back(spreading::unpack(
     4492          |                                                                                              ^
     4493    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'std::decay_t<U> hpx::util::detail::container_remapping::remap_container(container_mapping_tag<false, true>, M&&, T&&)
             ':
  >> 4494    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:533:81: error: '__T0' does not name a type
     4495      533 |                 val = spreading::unpack(
     4496          |                                                                                 ^
     4497    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'hpx::util::detail::spreading::spread_box<> hpx::util::detail::container_remapping::remap_container(container_mapping_
             tag<true, false>, M&&, T&&)':
  >> 4498    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:548:57: error: '__T0' does not name a type
     4499      548 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(val), val));
     4500          |                                                         ^~~~
     4501    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'void hpx::util::detail::container_remapping::remap(hpx::util::detail::strategy_traverse_tag, T&&, M&&, std::enable_if
             _t<is_effective_v<M, typename std::conditional<is_rvalue_reference_v<Container&&>, decltype (static_cast<typename std::remove_reference<decltype (* dec
             lval<Container>().begin())>::type&&>(* declval<Container>().begin())), decltype (* declval<Container>().begin())>::type> >*)':
  >> 4502    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:574:57: error: '__T0' does not name a type
     4503      574 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(element), element));
     4504          |                                                         ^~~~
     4505    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp: In fun
             ction 'void hpx::util::detail::format_value(std::ostream&, std::string_view, const format_join<Range>&)':
  >> 4506    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:34:
              error: '__T0' was not declared in this scope; did you mean '__y0'?
     4507      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4508          |                                  ^~~~
     4509          |                                  __y0
  >> 4510    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:38:
              error: template argument 1 is invalid
     4511      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4512          |                                      ^
  >> 4513    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 1 is invalid
     4514      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4515          |                                                 ^
  >> 4516    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 2 is invalid
  >> 4517    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:51:
              error: '<declaration error>' is not a template [-fpermissive]
     4518      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4519          |                                                   ^~
  >> 4520    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:73:
              error: expression list treated as compound expression in initializer [-fpermissive]
     4521      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4522          |                                                                         ^
     4523    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'decltype (rebind_container<hpx::util::detail::container_remapping::mapped_type_from_t<T, M> >(container)) hpx::util::
             detail::container_remapping::remap_container(container_mapping_tag<false, false>, M&&, T&&)':
  >> 4524    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:518:94: error: '__T0' does not name a type
     4525      518 |                 remapped.push_back(spreading::unpack(
     4526          |                                                                                              ^
     4527    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'std::decay_t<U> hpx::util::detail::container_remapping::remap_container(container_mapping_tag<false, true>, M&&, T&&)
             ':
  >> 4528    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:533:81: error: '__T0' does not name a type
     4529      533 |                 val = spreading::unpack(
     4530          |                                                                                 ^
     4531    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'hpx::util::detail::spreading::spread_box<> hpx::util::detail::container_remapping::remap_container(container_mapping_
             tag<true, false>, M&&, T&&)':
  >> 4532    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:548:57: error: '__T0' does not name a type
     4533      548 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(val), val));
     4534          |                                                         ^~~~
     4535    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp: In function 'void hpx::util::detail::container_remapping::remap(hpx::util::detail::strategy_traverse_tag, T&&, M&&, std::enable_if
             _t<is_effective_v<M, typename std::conditional<is_rvalue_reference_v<Container&&>, decltype (static_cast<typename std::remove_reference<decltype (* dec
             lval<Container>().begin())>::type&&>(* declval<Container>().begin())), decltype (* declval<Container>().begin())>::type> >*)':
  >> 4536    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/pack_traversal/detail/pack
             _traversal_impl.hpp:574:57: error: '__T0' does not name a type
     4537      574 |                 HPX_FORWARD(M, mapper)(HPX_FORWARD(decltype(element), element));
     4538          |                                                         ^~~~
     4539    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp: In lambda function:
  >> 4540    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:84: error: '__T0' was not declared in this scope; did you mean '__y0'?
     4541      273 |                             return hpx::execution::experimental::connect(
     4542          |                                                                                    ^
     4543          |                                                                                    __y0
  >> 4544    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:88: error: template argument 1 is invalid
     4545      273 |                             return hpx::execution::experimental::connect(
     4546          |                                                                                        ^
  >> 4547    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:91: error: expected '>' before '&&' token
     4548      273 |                             return hpx::execution::experimental::connect(
     4549          |                                                                                           ^
  >> 4550    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:91: error: expected '(' before '&&' token
  >> 4551    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:273:93: error: expected identifier before '>' token
     4552      273 |                             return hpx::execution::experimental::connect(
     4553          |                                                                                             ^
     4554    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp: In fun
             ction 'void hpx::util::detail::format_value(std::ostream&, std::string_view, const format_join<Range>&)':
  >> 4555    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:34:
              error: '__T0' was not declared in this scope; did you mean '__y0'?
     4556      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4557          |                                  ^~~~
     4558          |                                  __y0
  >> 4559    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:38:
              error: template argument 1 is invalid
     4560      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4561          |                                      ^
  >> 4562    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 1 is invalid
     4563      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4564          |                                                 ^
  >> 4565    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 2 is invalid
  >> 4566    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:51:
              error: '<declaration error>' is not a template [-fpermissive]
     4567      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4568          |                                                   ^~
  >> 4569    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:73:
              error: expression list treated as compound expression in initializer [-fpermissive]
     4570      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4571          |                                                                         ^
     4572    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp: In member function 'void hpx::when_all_vector_detail::when_all_vector_sender_impl<Sender>::when_all_vector_sender_type::operation_state
             <Receiver>::finish()':
  >> 4573    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:56: error: '__T0' was not declared in this scope; did you mean '__y0'?
     4574      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4575          |                                                        ^~~~
     4576          |                                                        __y0
  >> 4577    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:60: error: template argument 1 is invalid
     4578      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4579          |                                                            ^
  >> 4580    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:63: error: expected '>' before '&&' token
     4581      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4582          |                                                               ^~
  >> 4583    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:63: error: expected '(' before '&&' token
  >> 4584    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/execution/algorithms/when_
             all_vector.hpp:315:65: error: expected identifier before '>' token
     4585      315 |                                 values.push_back(HPX_MOVE(t.value()));
     4586          |                                                                 ^
     4587    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp: In fun
             ction 'void hpx::util::detail::format_value(std::ostream&, std::string_view, const format_join<Range>&)':
  >> 4588    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:34:
              error: '__T0' was not declared in this scope; did you mean '__y0'?
     4589      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4590          |                                  ^~~~
     4591          |                                  __y0
  >> 4592    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:38:
              error: template argument 1 is invalid
     4593      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4594          |                                      ^
  >> 4595    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 1 is invalid
     4596      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4597          |                                                 ^
  >> 4598    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 2 is invalid
  >> 4599    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:51:
              error: '<declaration error>' is not a template [-fpermissive]
     4600      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4601          |                                                   ^~
  >> 4602    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:73:
              error: expression list treated as compound expression in initializer [-fpermissive]
     4603      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4604          |                                                                         ^
     4605    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp: In fun
             ction 'void hpx::util::detail::format_value(std::ostream&, std::string_view, const format_join<Range>&)':
  >> 4606    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:34:
              error: '__T0' was not declared in this scope; did you mean '__y0'?
     4607      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4608          |                                  ^~~~
     4609          |                                  __y0
  >> 4610    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:320:38:
              error: template argument 1 is invalid
     4611      320 |                     using value_type = std::decay_t<decltype(elem)>;
     4612          |                                      ^
  >> 4613    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-1.9.1-ect2c3zuvuxqfo4yxtxzbgajxio4pvw6/include/hpx/modules/format.hpp:321:49:
              error: template argument 1 is invalid
     4614      321 |                     detail::formatter<value_type>::call(os, spec, &elem);
     4615          |                                     

```

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1. Compile HPX on Grace

## Specifications

... Please describe your environment

  - HPX Version: 1.9 and 1.10
  - Platform (compiler, OS): Linux, gcc 12 or gcc 13

Ok, now I get this one here

 [ 98%] Built target unitiger
     3053    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
             ect.hpp: In function 'std::vector<Future> hpx::lcos::detail::return_result_type(hpx::future<std::vector<hpx::future<std::vector<Future> > > >)':
  >> 3054    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
             ect.hpp:369:65: error: '__T1' was not declared in this scope; did you mean '__T0'?
     3055      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
     3056          |                                                                 ^
     3057          |                                                                 __T0
  >> 3058    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
             ect.hpp:369:69: error: template argument 1 is invalid
     3059      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
     3060          |                                                                     ^
  >> 3061    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
             ect.hpp:369:72: error: expected '>' before '&&' token
     3062      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
     3063          |                                                                        ^
  >> 3064    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
             ect.hpp:369:72: error: expected '(' before '&&' token
     3065      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
     3066          |                                                                        ^
     3067          |                                                                        (
  >> 3068    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
             ect.hpp:369:74: error: expected identifier before '>' token
     3069      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
     3070          |                                                                          ^
  >> 3071    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
             ect.hpp:369:84: error: expected ')' before ';' token
     3072      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
     3073          |                                                                                    ^
     3074          |                                                                                    )
  >> 3075    make[2]: *** [CMakeFiles/octotiger.dir/build.make:107: CMakeFiles/octotiger.dir/frontend/init_methods.cpp.o] Error 1
     3076    make[2]: Leaving directory '/ram/tmp/diehlpk/spack-stage/spack-stage-octotiger-master-xjjnhchcavp7jcklkoups63f62rj5muz/spack-src/spack-build'
  >> 3077    make[1]: *** [CMakeFiles/Makefile2:249: CMakeFiles/octotiger.dir/all] Error 2
     3078    make[1]: Leaving directory '/ram/tmp/diehlpk/spack-stage/spack-stage-octotiger-master-xjjnhchcavp7jcklkoups63f62rj5muz/spack-src/spack-build'
  >> 3079    make: *** [Makefile:139: all] Error 2

> Ok, now I get this one here
> 
> ```
>  [ 98%] Built target unitiger
>      3053    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
>              ect.hpp: In function 'std::vector<Future> hpx::lcos::detail::return_result_type(hpx::future<std::vector<hpx::future<std::vector<Future> > > >)':
>   >> 3054    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
>              ect.hpp:369:65: error: '__T1' was not declared in this scope; did you mean '__T0'?
>      3055      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
>      3056          |                                                                 ^
>      3057          |                                                                 __T0
>   >> 3058    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
>              ect.hpp:369:69: error: template argument 1 is invalid
>      3059      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
>      3060          |                                                                     ^
>   >> 3061    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
>              ect.hpp:369:72: error: expected '>' before '&&' token
>      3062      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
>      3063          |                                                                        ^
>   >> 3064    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
>              ect.hpp:369:72: error: expected '(' before '&&' token
>      3065      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
>      3066          |                                                                        ^
>      3067          |                                                                        (
>   >> 3068    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
>              ect.hpp:369:74: error: expected identifier before '>' token
>      3069      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
>      3070          |                                                                          ^
>   >> 3071    /vast/home/diehlpk/spack/opt/spack/linux-rhel9-neoverse_v2/gcc-13.2.0/hpx-master-bkwrmghaxyzwf2ul46k6alhh726km6xk/include/hpx/collectives/broadcast_dir
>              ect.hpp:369:84: error: expected ')' before ';' token
>      3072      369 |                 std::vector<Result> t = HPX_MOVE(f.get());
>      3073          |                                                                                    ^
>      3074          |                                                                                    )
>   >> 3075    make[2]: *** [CMakeFiles/octotiger.dir/build.make:107: CMakeFiles/octotiger.dir/frontend/init_methods.cpp.o] Error 1
>      3076    make[2]: Leaving directory '/ram/tmp/diehlpk/spack-stage/spack-stage-octotiger-master-xjjnhchcavp7jcklkoups63f62rj5muz/spack-src/spack-build'
>   >> 3077    make[1]: *** [CMakeFiles/Makefile2:249: CMakeFiles/octotiger.dir/all] Error 2
>      3078    make[1]: Leaving directory '/ram/tmp/diehlpk/spack-stage/spack-stage-octotiger-master-xjjnhchcavp7jcklkoups63f62rj5muz/spack-src/spack-build'
>   >> 3079    make: *** [Makefile:139: all] Error 2
> ```

Let me have a look. It's the same nvcc problem as before.

--- a/libs/full/collectives/include/hpx/collectives/broadcast_direct.hpp
+++ b/libs/full/collectives/include/hpx/collectives/broadcast_direct.hpp
@@ -361,12 +361,11 @@ namespace hpx { namespace lcos {
             hpx::future<std::vector<hpx::future<std::vector<Result>>>> r)
         {
             std::vector<Result> res;
-            std::vector<hpx::future<std::vector<Result>>> fres =
-                HPX_MOVE(r.get());
+            std::vector<hpx::future<std::vector<Result>>> fres = r.get();
 
             for (hpx::future<std::vector<Result>>& f : fres)
             {
-                std::vector<Result> t = HPX_MOVE(f.get());
+                std::vector<Result> t = f.get();
                 res.reserve(res.capacity() + t.size());
                 std::move(t.begin(), t.end(), std::back_inserter(res));
             }

--- a/libs/full/collectives/include/hpx/collectives/reduce_direct.hpp
+++ b/libs/full/collectives/include/hpx/collectives/reduce_direct.hpp
@@ -209,7 +209,7 @@ namespace hpx { namespace lcos {
             Result operator()(
                 hpx::future<std::vector<hpx::future<Result>>> r) const
             {
-                std::vector<hpx::future<Result>> fres = HPX_MOVE(r.get());
+                std::vector<hpx::future<Result>> fres = r.get();
 
                 HPX_ASSERT(!fres.empty());
 

